docs/OnTrack/Deployment/Google Cloud/Overview.md

```diff
@@ -30,40 +36,121 @@ is a small, initial footprint that has a publicly facing network and accessible
-server, app server, doubtfire-web, mariadb, nginx).
-
-## Aims
-
-- Deploy a student-run version of OnTrack hosted on Google Cloud. This version will be running in
-  isolation, independently of the Deakin University version.
-- Create a CI/CD pipeline that automates the building, deployment, and validation of a of the Thoth
-  Tech OnTrack to Google Cloud.
+used docker compose to deploy the images for the components required to run Doubtfire (api server,
+app server, doubtfire-web, mariadb, nginx).
 
 ## Initial stages
 
 Initially, the Google Cloud team had spent time understanding GCP, Docker, and the Doubtfire
-deployment. We ran several tests locally prior to deploying a server instance in GCP and deploying
-Doubtfire.
+deployment.
+
+### Tests via localhost
+
+Prior to deploying to GCP, we ran several tests locally (localhost) on our own workstations to
+determine the configuration changes required deploy Doubtfire successfully. On our individual
+workstations, we cloned the
+[Doubtfire-deploy-GCP repository](https://github.com/thoth-tech/doubtfire-deploy-GCP) and modified
+the necessary files. We then used docker compose and Docker to run and deploy containers.
+
+![doubtfire-localhost-compose](https://github.com/thoth-tech/documentation/tree/main/docs/OnTrack/Deployment/Google%20Cloud/assets/doubtfire-localhost-compose.png "docker compose output")
+
+Success! we have Docker containers running locally.
+
+![doubtfire-localhost-docker](https://github.com/thoth-tech/documentation/tree/main/docs/OnTrack/Deployment/Google%20Cloud/assets/doubtfire-localhost-docker.png "Docker containers running")
+
+Success again! We have OnTrack being hosted locally and is accessible via https://localhost:443 .
+
+![doubtfire-localhost](https://github.com/thoth-tech/documentation/tree/main/docs/OnTrack/Deployment/Google%20Cloud/assets/doubtfire-localhost.png "Doubtfire running on localhost")
+
+### Google Compute Engine instance
+
+Once we determined the configuration changes required to be able to run locally (localhost), we then
+needed to determine how to create and deploy a Compute Engine server instance in GCP that we could
+use to deploy Doubtfire.
+
+We started with a small, initial footprint and deployed a basic virtual machine (VM) instance with
+following details -
+
+- Name: instance-1
+- Zone: australia-southeast2-a
+- Machine type: e2-micro
+- CPU platform: Intel Broadwell
+- OS Image: Centos 7 (Image name centos-7-v20220406)
+- Boot disk: 20GB SCSI (Balanced persistent disk, Google-managed encryption)
+- Network interfaces: nic0 (with internal & external IP addresses)
+- Firewalls: HTTP, HTTPS enabled
+- GPUs/Display device: None, disabled
+
+You can find the equivalent REST for our original instance from Google Compute Engine
+[here](https://github.com/thoth-tech/documentation/tree/main/docs/OnTrack/Deployment/Google%20Cloud/assets/gcp-instance-rest.json).
```

Comment:
**question:** Do we have this file in the https://github.com/thoth-tech/doubtfire-deploy-GCP repo? 🤔

Reply:
I removed it for now as the lintliner kept throwing me an EOF error, even after adding an extra line to the end of the file and vaildating that the JSON syntax was correct. Not really important, it was just a nice to have.